.ci/lint

```diff
@@ -41,52 +43,117 @@ pipeline {
                 archiveArtifacts artifacts: 'file-list-status.log,file-list-changed.log,file-list-deleted.log,file-list-experiment.log'
             }
         }
-        stage('linting') {
-            failFast false
+        stage('linting chg') {
             parallel {
-                stage('salt linting') {
+                stage('lint salt') {
                     when {
                         expression { return readFile('file-list-changed.log') =~ /(?i)(^|\n)(salt\/.*\.py|setup\.py)\n/ }
                     }
                     steps {
                         sh '''
                         eval "$(pyenv init - --no-rehash)"
-                        grep -Ei '^salt/.*\\.py$|^setup\\.py$' file-list-changed.log | xargs -r '--delimiter=\\n' tox -e pylint-salt | tee pylint-report-salt.log
+                        # tee makes the exit/return code always 0
+                        grep -Ei '^salt/.*\\.py$|^setup\\.py$' file-list-changed.log | (xargs -r '--delimiter=\\n' tox -e pylint-salt;echo"$?" >  pylint-salt-chg.exit) | tee pylint-report-salt.log
```

Comment:
https://jenkinsci.saltstack.com/job/pr-lint/job/PR-50583/6/

`echo"$?"` This needs to be fixed.